dhall/src/Dhall/TypeCheck.hs

```diff
@@ -3535,6 +3544,51 @@ prettyTypeMessage (MissingField k expr0) = ErrorMessages {..}
         txt0 = insert k
         txt1 = insert expr0
 
+prettyTypeMessage (ProjectionTypeMismatch k expr0 expr1) = ErrorMessages {..}
+  where
+    short = "Projection type mismatch"
```

Comment:
E.g. `AnnotMismatch` includes a diff in the short message. I did the same in

https://github.com/dhall-lang/dhall-haskell/blob/4788b886956588b9a081bb114d083a3b88857855/dhall/src/Dhall/TypeCheck.hs#L3508-L3512

```
⊢ {a = 1, b = {=}, c = True}.({a : Bool, b : Text})

Error: Selector field types don't match record field types

{ a : - Bool
      + Natural
, b : - Text
      + { … : … }
}

1: {a = 1, b = {=}, c = True}.({a : Bool, b : Text})
```

Not sure if that's a great idea though.